projects/packages/waf/tests/php/test-waf-compatibility.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Compatibility test suite.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+use Automattic\Jetpack\Waf\Waf_Compatibility;
+
+/**
+ * Compatibility test suite.
+ */
+class WafCompatibilityTest extends PHPUnit\Framework\TestCase {
+
+	/**
+	 * Test Waf_Compatibility::merge_ip_allow_lists().
+	 */
+	public function testMergeIPAllowLists() {
+		$waf_allow_list         = "1.1.1.1,2.2.2.2\n3.3.3.3-4.4.4.4";
+		$brute_force_allow_list = array(
+			(object) array(
+				'ip_address' => '5.5.5.5',
+				'range'      => false,
+			),
+			(object) array(
+				'range'      => true,
+				'range_low'  => '6.6.6.6',
+				'range_high' => '7.7.7.7',
+			),
+			(object) array(
+				'ip_address' => '8.8.8.8',
+			),
+		);
```

Comment:
I wonder if it would be worth adding a test for a case like this:
```php
		$waf_allow_list         = "1.1.1.1,2.2.2.2\n3.3.3.3-4.4.4.4";
		$brute_force_allow_list = array(
			(object) array(
				'ip_address' => '1.1.1.1',
				'range'      => false,
			),
			(object) array(
				'range'      => true,
				'range_low'  => '3.3.3.3',
				'range_high' => '4.4.4.4',
			),
			(object) array(
				'ip_address' => '8.8.8.8',
			),
		);
```
The question is whether the result should be like
```
$expected_result = "1.1.1.1,2.2.2.2\n3.3.3.3-4.4.4.4\n1.1.1.1\n3.3.3.3-4.4.4.4\n8.8.8.8";
```
or if it should have eliminated the duplicate entries.

Reply:
It currently will not remove any duplicate entries in the merged IP list, but including this test might help define that expectation. I'll add it! :+1:

Reply:
Added the test, good thing you pointed this out as well because the test file needed to be moved into the recently added `unit` directory :+1:

Added via https://github.com/Automattic/jetpack/pull/28633/commits/6c68a83dd9b3b279275f1749708d1ebe180d5946